src/agent/onefuzz-task/src/tasks/task_logger.rs

```diff
@@ -150,25 +154,24 @@ impl LogWriter<BlobLogWriter> for BlobLogWriter {
         let result = blob_client
             .append_block(data_stream)
             .condition_max_size(self.max_log_size)
-            .execute()
+            .into_future()
             .await;
 
         match result {
             Ok(_r) => Ok(WriteLogResponse::Success),
             Err(e) => match e.downcast_ref::<HttpError>() {
-                Some(HttpError::StatusCode { status: s, body: b }) => {
-                    if s == &StatusCode::PRECONDITION_FAILED
-                        && b.contains("MaxBlobSizeConditionNotMet")
+                Some(herr) => match herr.status() {
+                    StatusCode::PreconditionFailed
+                        if herr.error_code() == Some("MaxBlobSizeConditionNotMet") =>
                     {
                         Ok(WriteLogResponse::MaxSizeReached)
-                    } else if s == &StatusCode::CONFLICT && b.contains("BlockCountExceedsLimit") {
+                    }
+                    StatusCode::Conflict if herr.error_code() == Some("BlockCountExceedsLimit") => {
```

Comment:
This (and the above) are different checks, but probably preferable. Maybe worth double-checking that by using an equality check (instead of substring match), we aren't excluding any error codes that we were previously accepting.

Reply:
Going by the [Azure Blob Storage error codes](https://docs.microsoft.com/en-us/rest/api/storageservices/blob-service-error-codes) article, these are not substrings of any other error codes.